types/svg-parser/index.d.ts

```diff
@@ -1,14 +1,24 @@
-// Type definitions for svg-parser 1.0
+// Type definitions for svg-parser 2.0
 // Project: https://gitlab.com/Rich-Harris/svg-parser#README
 // Definitions by: mrmlnc <https://github.com/mrmlnc>
+//                 Joel Shinness <https://github.com/JoelCodes>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
-export interface Node {
-    name: string;
-    attributes: Record<string, string | number>;
-    children: Node[];
+export interface TextNode {
+    type: 'text';
+    value?: string | boolean | number;
+}
+
+export interface ElementNode {
+    type: string;
+    tagName?: string;
+    properties?: {};
+    children: Array<Node | string>;
+    value?: string;
     metadata?: string;
 }
 
-export function parse(content: string): Node;
+export type Node = TextNode | ElementNode;
```

Comment:
Unfortunately, now this type is not fully correct.

The second version of the packages is always returns `root` node:

```js
{ type: 'root', children: [] }
```

https://github.com/Rich-Harris/svg-parser/blob/734f11b37397eafed09a4688a505972d9f5613b7/src/index.js#L231-L234
https://github.com/Rich-Harris/svg-parser/blob/master/test/samples/empty-svg/output.json

We can make this place better by always returning the `RootNode` node. Something like it:

```js
export type RootNode {
  type: 'root';
  children: TextNode | ElementNode;
}
```

Reply:
That's a good catch!  Thanks!